exec/cnex/support.c

```diff
@@ -0,0 +1,212 @@
+#ifdef _WIN32
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+
+#include <iso646.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "bytecode.h"
+#include "module.h"
+#include "string.h"
+#include "support.h"
+#include "util.h"
+
+#ifdef _WIN32
+#include <io.h>
+#define access _access
+#define R_OK 04
+#define PATHSEP ';'
+#define strdup _strdup
+#else
+#include <unistd.h>
+#define PATHSEP ':'
+#endif
+
+
+static TPath *neon_paths;
+
+static void path_pushPath(const char *source_path)
+{
+    TPath *p = neon_paths;
+
+    TPath *n = malloc(sizeof(TPath));
+    size_t len = strlen(source_path);
+    n->path = malloc(len + 1);
+    memcpy(n->path, source_path, len);
+    n->path[len] = '\0';
+    n->next = NULL;
+
+    // If we haven't initialized the Neon paths yet, go ahead and set n to the root
+    // and return.
+    if (p == NULL) {
+        neon_paths = n;
+        return;
+    }
+
+    // Jump to the end of the list; to add this path onto our search paths.
+    while (p && p->next) {
+        p = p->next;
+    }
+
+    // Add this new path to the paths.
+    p->next = n;
+}
+
+void path_freePaths(TPath *paths)
+{
+    TPath *n = paths->next;
```

Comment:
Why does this start at the second item in the list? Wouldn't it be simpler to start at the first? Then, along with simplifying the code, it won't segfault if you pass in an empty list.

Reply:
Refactored; made more simple.  Also added a call *TO* this this function to clean up the memory!